src/layers/heatmap-layer/heatmap-layer.js

```diff
@@ -150,26 +152,26 @@ class HeatmapLayer extends MapboxGLLayer {
   };
 
   datasetSelector = config => config.dataId;
-  isVisibleSelector = config => config.isVisible;
+  columnsSelector = config => pointColResolver(config.columns);
   visConfigSelector = config => config.visConfig;
   weightFieldSelector = config => config.weightField ? config.weightField.name : null;
   weightDomainSelector = config => config.weightDomain;
 
   computeHeatmapConfiguration = createSelector(
     this.datasetSelector,
-    this.isVisibleSelector,
+    this.columnsSelector,
     this.visConfigSelector,
     this.weightFieldSelector,
     this.weightDomainSelector,
 
-    (datasetId, isVisible, visConfig, weightField, weightDomain) => {
+    (datasetId, columns, visConfig, weightField, weightDomain) => {
 
-      const layer = {
+      return {
         type: 'heatmap',
         id: this.id,
-        source: datasetId,
+        source: `${datasetId}-${columns}`,
```

Comment:
Setting source to `datasetId` plus `columns` so that multiple heatmap based on the same dataset but different column can be created